src/utils/string.hpp

```diff
@@ -459,4 +460,31 @@ inline std::string_view Substr(const std::string_view string, size_t pos = 0, si
   return string.substr(pos, len);
 }
 
+/**
+ * Convert a double value to a string representation.
+ * Precision of converted value is 16.
+ * Function also removes trailing zeros after the dot.
+ *
+ * @param value The double value to be converted.
+ *
+ * @return The string representation of the double value.
+ *
+ * @throws None
+ */
+inline std::string DoubleToString(const double value) {
+  static const int PRECISION = 15;
+  static const int STR_LENGTH = PRECISION + 1;  // 1 for dot
```

Comment:
We shouldn't need to calculate this ourselves.
Using `std::precision` with `std::fixed` means that we are defining the number of decimal places, not the overall number of digits. So the length is actually different from `STR_PRECISON`.
We can just use the whole stringstream's string_view.

Reply:
https://godbolt.org/z/T555Kr654
`setprecision` sets count of numbers after dot + dot symbol.
So this case will be broken.
`EXPECT_EQ(DoubleToString(1234567.01234567891234567), "1234567.01234567");`

Reply:
Updated.